### DISCUSSION TITLE
Celo data encryption key.

### DISCUSSION THREAD

**Discussion by Allen-Muhani** [celo-org/social-connect#247] [Help (Q&A)]
How do you support saving the DEK on the accounts smart contracts with 0 as the net balance on all tokens?? I thought of doing a cloud function to send like 0.05 celo/cUSD for this purpose could there be a better option out there

**Comment by arthurgousset** [comment]
Would you be able to dig into @Allen-Muhani's question and see if you can help him @therealharpaljadeja ?
Thanks 🙏

**Comment by therealharpaljadeja** [comment]
If I understood the question right, you are asking how we can let users set a DEK for their Account when they are just starting and don't have any balance.

There are multiple ways to tackle this. I would recommend doing meta-tx, where you sponsor this transaction for the user.

One very intuitive way to implement this is using Openzeppelin's Relay service. Here's a link to that

https://www.youtube.com/watch?v=mhAUmULLV44

Do let me know If I understood your question correctly. Happy to help!

**Reply by Allen-Muhani** [reply]
Yes. you have understood the question. Basically, the onboarding process needs to be gasless for users with nothing/ no balance on their accounts.

**Comment by Allen-Muhani** [comment]
If I have understood the tutorial correctly. I do need to alter the smart contracts to have the necessary openzeppeling contract dependencies. However in this case the accounts smart contract is managed by cLabs and not me.
https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/common/Accounts.sol

Was wondering if I could get help around komenci as they seem to already have it for this specific purpose.
https://docs.celo.org/protocol/identity/smart-contract-accounts#separation-of-signer-and-payer

**Reply by therealharpaljadeja** [reply]
May I know the purpose of setting the DEK? 

If you are asking for SocialConnect, then the issuer needs to have a DEK, and it is expected that the issuer is an entity which has Celo and cUSD tokens for the process.

User can register on SocialConnect without any balance because the issuer takes the overhead of paying for gas and paying for ODIS quota.

**Reply by Allen-Muhani** [reply]
I am using it for comment encryption

**Reply by therealharpaljadeja** [reply]
Currently, there is no way to set DEK without gas, since msg.sender is being used in Accounts.sol

**Comment by Allen-Muhani** [comment]
Cool. Let me create one that's quasi-free.

Thanks for offering your time to help.
